File: app_tests/java/VulnerableApp.java
import java.sql.*;
import java.util.Scanner;

public class VulnerableApp {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter user ID: ");
        String userId = scanner.nextLine();
        
        // SQL Injection vulnerability
        String query = "SELECT * FROM users WHERE id = " + userId;
        
        // Command injection vulnerability  
        try {
            Runtime.getRuntime().exec("ls -la " + userId);
        } catch (Exception e) {
            e.printStackTrace();
        }
        
        // Hardcoded password
        String password = "admin123";
        
        // Weak random
        java.util.Random rand = new java.util.Random();
        int token = rand.nextInt();
        
        System.out.println("Query: " + query);
    }
}

File: app_tests/javascript/vulnerable_app.js
const express = require('express');
const { exec } = require('child_process');
const crypto = require('crypto');

const app = express();

app.get('/search', (req, res) => {
    const userInput = req.query.q;
    
    // Command injection vulnerability
    exec(`find . -name "${userInput}"`, (error, stdout, stderr) => {
        if (error) {
            console.error(error);
            return;
        }
        res.send(stdout);
    });
});

app.get('/user/:id', (req, res) => {
    const userId = req.params.id;
    
    // SQL injection vulnerability
    const query = `SELECT * FROM users WHERE id = ${userId}`;
    console.log(query);
    
    // Hardcoded secret
    const jwtSecret = "super_secret_key_123";
    
    // Weak random
    const token = Math.random().toString(36);
    
    // eval usage - code injection
    const userCode = req.query.code;
    if (userCode) {
        eval(userCode);
    }
    
    res.json({ userId, token });
});

app.listen(3000, () => {
    console.log('Server running on port 3000');
});
